src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -602,12 +603,22 @@ private GraphQLObjectType getActualType(GraphQLObjectType type) {
     }
 
     private boolean isConnection(AccessibleObject obj, GraphQLOutputType type) {
+        ConnectionTypeValidator validator = new ConnectionTypeValidator();
         if (type instanceof GraphQLNonNull) {
             type = (GraphQLOutputType) ((GraphQLNonNull) type).getWrappedType();
         }
         final GraphQLOutputType actualType = type;
-        return obj.isAnnotationPresent(GraphQLConnection.class) &&
-                actualType instanceof GraphQLList && TYPES_FOR_CONNECTION.stream().anyMatch(aClass -> aClass.isInstance(((GraphQLList) actualType).getWrappedType()));
+
+        boolean isValidGraphQLTypeForConnection = obj.isAnnotationPresent(GraphQLConnection.class) &&
+                actualType instanceof GraphQLList && TYPES_FOR_CONNECTION.stream().anyMatch(aClass ->
+                aClass.isInstance(((GraphQLList) actualType).getWrappedType()));
+
+        if(isValidGraphQLTypeForConnection) {
+            validator.validate(obj);
```

Comment:
it throws an exception? why not to handle it here?

Reply:
I want the exception to be thrown, because it suggest that the `@GraphQLConnection` is not used properly, so it must be fixed in order to build the schema correclty